pingcap/tidb-binlog
# Pump batch pushing binlog

### What problem does this PR solve? 
improve pump throughput pushing binlog to kafka

### What is changed and how it works?
benefit from batch push to kafka.

a ProduceRequest can contains many msg to the broken
internally, AsyncProducer will do this(SyncProducer is a wrapper of(AsyncProducer),
set the Retry to a pretty high value to avoid not continuity and control
when to quit ourself,you may want to take a look of
Shopify/sarama#838

### Check List 

Tests 


 - Manual test (add detailed scripts or steps below)
continual run integrate test and restart kafka per 2 minutes, make sure test never fail

Code changes

 

Side effects


Related changes
